tmt/base.py

```diff
@@ -1796,7 +1796,12 @@ def _initialize_worktree(self) -> None:
 
         # Sync metadata root to the worktree
         self.debug(f"Sync the worktree to '{self.worktree}'.", level=2)
-        self.run(Command("rsync", "-ar", "--exclude", ".git", f"{tree_root}/", self.worktree))
+        # If we're in a git repository, we just clone it so that we correctly
+        # honor .gitignore, etc.
+        if os.path.isdir(f"{tree_root}/.git"):
+            self.run(Command("git", "clone", "-s", tree_root, self.worktree))
```

Comment:
As I wrote above, that discards local uncommitted modifications. So it changes behaviour in an IMHO unexpected way.

Reply:
Well, the source tree is the very thing that actually has the tests, so that seems rather essential. There could be some heuristics to only copy the fmf files/plans/subdirs which have tests, but tests currently can assume that they have access to the source tree (we depend on that to get auxiliary files, node modules, etc.)

Reply:
Honoring .gitignore would certainly go a long way to fix this.

Reply:
OK reworked to use `git ls-files`.